engine/src/test/java/org/camunda/bpm/engine/test/persistence/ConnectionPersistenceExceptionTest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.engine.test.persistence;
+
+import org.apache.ibatis.datasource.pooled.PooledDataSource;
+import org.assertj.core.api.ThrowableAssert;
+import org.camunda.bpm.engine.IdentityService;
+import org.camunda.bpm.engine.identity.User;
+import org.camunda.bpm.engine.impl.cfg.ProcessEngineConfigurationImpl;
+import org.camunda.bpm.engine.impl.db.sql.DbSqlSessionFactory;
+import org.camunda.bpm.engine.impl.test.RequiredDatabase;
+import org.camunda.bpm.engine.impl.util.PropertiesUtil;
+import org.camunda.bpm.engine.test.ProcessEngineRule;
+import org.camunda.bpm.engine.test.util.ProvidedProcessEngineRule;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Properties;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.camunda.bpm.engine.impl.util.ExceptionUtil.PERSISTENCE_CONNECTION_ERROR_CLASS;
+
+@RequiredDatabase(excludes = { DbSqlSessionFactory.H2 })
+public class ConnectionPersistenceExceptionTest {
```

Comment:
🔧 This test class tests that the behavior we use (fetching the cause of the cause of a failing MyBatis call) is valid. However, we don't test yet that we use the behavior. We should add a test case to the REST API ensuring that we log SQLExceptions of class `08` on level ERROR. You can mock the result of a REST call easily to do this and use the log watcher to assert the logged statements for specific loggers.

Reply:
❓ Under the **limitations of Scope** section, it mentions that automated testing of this functionality is out of scope. Also, it has been tested manually and verified that the logging does occur for class 8 connection exceptions. Would it be ok if we skip this test for now?

Reply:
👍 Good point. Yeah, absolutely fair. I read it but did not put this into context here. Let's skip that 👍

Reply:
The scope limitation is about H2, not so much about adding a test for the REST API. If Tobi's proposal is easy to realize, I would appreciate that. If not, we could also skip it.

Reply:
❓ Is it possible to add this test in the form of technical debt after the feature is merged due to time restrictions and the upcoming code-freeze?

Reply:
Sure. Please create a follow-up ticket for that.

Reply:
I've created [this](https://github.com/camunda/camunda-bpm-platform/issues/3762) task as a follow-up to continue with the addition of the test.